test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+            testutil_check(system(buf));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+
+    /* Done processing all files. Close backup cursor. */
+    testutil_check(backup_cur->close(backup_cur));
+    finalize_files(flist, count);
+    free(tmp);
+    /*! [incremental backup using block transfer]*/
+}
+
+static void
+add_data_validate_backups(WT_SESSION *session)
+{
+    int i, j;
+    printf("Adding initial data\n");
+    add_work(session, 0, 0, uri, uri2);
+
+    printf("Taking initial backup\n");
+    take_full_backup(session, 0);
+
+    testutil_check(session->checkpoint(session, NULL));
+
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: adding data\n", i);
+        /* For each iteration we may add work and checkpoint multiple times. */
+        for (j = 0; j < i; j++) {
+            add_work(session, i, j, uri, uri2);
+            testutil_check(session->checkpoint(session, NULL));
+        }
+
+        /*
+         * The full backup here is only needed for testing and comparison purposes. A normal
+         * incremental backup procedure would not include this.
+         */
+        printf("Iteration %d: taking full backup\n", i);
+        take_full_backup(session, i);
+        /*
+         * Taking the incremental backup also calls truncate to archive the log files, if the copies
+         * were successful. See that function for details on that call.
+         */
+        printf("Iteration %d: taking incremental backup\n", i);
+        take_incr_backup(session, i);
+
+        printf("Iteration %d: dumping and comparing data\n", i);
+        testutil_check(compare_backups(i, uri));
+    }
+}
+
+/*
+ * Remove all the records, take backups and validate
+ */
+static void
+remove_all_records_validate(WT_SESSION *session)
+{
+    int i, j;
+
+    remove_work(session, 0, 0);
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: Removing data\n", i);
+        for (j = 0; j < i; j++) {
+            remove_work(session, i, j);
+        }
+    }
+    take_full_backup(session, i);
+
+    take_incr_backup(session, i);
+
+    printf("Dumping and comparing data\n");
+    testutil_check(compare_backups(i, uri));
+}
+
+static void
+drop_old_add_new_objects(WT_CONNECTION *wt_conn)
+{
+    WT_CURSOR *backup_cur;
+    WT_SESSION *session;
+    int ret, i, j;
+    char buf[1024];
+
+    testutil_check(wt_conn->close(wt_conn, NULL));
+    testutil_check(wiredtiger_open(home, NULL, CONN_CONFIG, &wt_conn));
+    testutil_check(wt_conn->open_session(wt_conn, NULL, NULL, &session));
+    testutil_check(session->create(session, uri3, "key_format=S,value_format=S"));
+
+    testutil_check(session->open_cursor(session, "backup:", NULL, NULL, &backup_cur));
+    testutil_check(backup_cur->close(backup_cur));
+
+    testutil_check(session->drop(session, uri2, "force"));
+
+    new_object = true;
+
+    for (i = ITERATIONS_MULTIPLIER + 1; i < ITERATIONS_MULTIPLIER * 2; i++) {
```

Comment:
The loop bounds are not intuitive and confusing. They feel like they're fragile, but I really don't know what you're expressing here. So either it should be clearly commented or think about whether it could be simplified. What does the loop give us? It seems like you could just do the drop and create and then add data to uri3 once and that would be sufficient.

Reply:
I have updated the code to remove irrelevant loops and added a global variable `"counter"` to drive this.
Note - The max value of `counter` would be MAX_ITERATIONS.